src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathMapperTest.java

```diff
@@ -786,6 +786,169 @@ public void testFourthCaseGroup() throws Exception {
+                .getNextSibling()
+                .getNextSibling();
+        final DetailAST expectedExprNode = expectedAssignNode.getFirstChild();
+        final DetailAST expectedNumIntNode = expectedExprNode.getFirstChild();
+        final DetailAST[] expected = {expectedAssignNode, expectedExprNode, expectedNumIntNode};
+        assertArrayEquals("Result nodes differ from expected", expected, actual);
+    }
+
+    @Test
+    public void testQueryElementFollowingMethodDef() throws Exception {
+        final String xpath = "//PACKAGE_DEF/following::METHOD_DEF";
+        final RootNode rootNode = getRootNode("InputXpathMapperAst.java");
+        final DetailAST[] actual = convertToArray(getXpathItems(xpath, rootNode));
+        final DetailAST expectedMethodDefNode = getSiblingByType(rootNode.getUnderlyingNode(),
+                TokenTypes.CLASS_DEF)
+                .findFirstToken(TokenTypes.OBJBLOCK)
+                .findFirstToken(TokenTypes.METHOD_DEF);
+        final DetailAST[] expected = {expectedMethodDefNode.getNextSibling(),
+            expectedMethodDefNode};
+        assertArrayEquals("Result nodes differ from expected", expected, actual);
+    }
+
+    @Test
+    public void testQueryElementNoFollowing() throws Exception {
+        final String xpath = "//CLASS_DEF/following::*";
+        final RootNode rootNode = getRootNode("InputXpathMapperAst.java");
+        final DetailAST[] actual = convertToArray(getXpathItems(xpath, rootNode));
+        assertEquals("Invalid number of nodes", 0, actual.length);
+    }
+
+    @Test
+    public void testQueryElementPrecedingSibling() throws Exception {
+        final String xpath = "//VARIABLE_DEF[./IDENT[@text='array']]/preceding-sibling::*";
+        final RootNode rootNode = getRootNode("InputXpathMapperAst.java");
+        final DetailAST[] actual = convertToArray(getXpathItems(xpath, rootNode));
+        final DetailAST expectedVariableDefNode1 = getSiblingByType(rootNode.getUnderlyingNode(),
+                TokenTypes.CLASS_DEF)
+                .findFirstToken(TokenTypes.OBJBLOCK)
+                .findFirstToken(TokenTypes.METHOD_DEF)
+                .findFirstToken(TokenTypes.SLIST)
+                .findFirstToken(TokenTypes.VARIABLE_DEF);
+        final DetailAST expectedSemiNode1 = expectedVariableDefNode1.getNextSibling();
+        final DetailAST expectedVariableDefNode2 = expectedSemiNode1.getNextSibling();
+        final DetailAST expectedSemiNode2 = expectedVariableDefNode2.getNextSibling();
+        final DetailAST[] expected = {expectedSemiNode2, expectedSemiNode1,
+            expectedVariableDefNode2, expectedVariableDefNode1};
+        assertArrayEquals("Result nodes differ from expected", expected, actual);
+    }
+
+    @Test
+    public void testQueryElementPrecedingSiblingVariableDef() throws Exception {
+        final String xpath = "//VARIABLE_DEF[./IDENT[@text='array']]/preceding-sibling::"
+                + "VARIABLE_DEF";
```

Comment:
please make it:
```
        final String xpath = "//VARIABLE_DEF[./IDENT[@text='array']]"
 + "/preceding-sibling::VARIABLE_DEF";
```
to cut by xpath element, it will be much easy to read.

Reply:
done